staging/kos/pkg/controllers/ipam/ipam.go

```diff
@@ -247,20 +253,16 @@ func (ctlr *IPAMController) Run(stopCh <-chan struct{}) error {
 		ctlr.OnLockCreate,
 		ctlr.OnLockUpdate,
 		ctlr.OnLockDelete})
-	go ctlr.lockInformer.Run(stopCh)
-	go ctlr.netattInformer.Run(stopCh)
-	go ctlr.subnetInformer.Run(stopCh)
-	glog.V(2).Infof("Informer Runs forked\n")
+
 	if !k8scache.WaitForCacheSync(stopCh, ctlr.subnetInformer.HasSynced, ctlr.lockInformer.HasSynced, ctlr.netattInformer.HasSynced) {
-		return fmt.Errorf("Caches failed to sync")
+		panic("informers' caches failed to sync")
```

Comment:
We panic so that the whole controller crashes and is restarted (including this controller). Returning an error is not viable as this code is run by its own, dedicated goroutine. panic instead of using `os.Exit` so that deferred functions can execute. The con of the approach is that other controllers which were running fine are stopped and restarted. I'm not done studying the upstream controller manager --- I'd like to understand how partial failures and restarts are handled. Maybe something like starting controllers with https://godoc.org/k8s.io/apimachinery/pkg/util/wait#Forever could be used.